src/consumer.rs

```diff
@@ -146,17 +136,105 @@ impl PartitionConsumerJS {
         &self,
         offset: OffsetWrapper,
     ) -> Result<PartitionConsumerIterator, FluvioError> {
-        if let Some(client) = &self.inner {
-            let stream = client.stream(offset.0).await?;
-            let mut iterator = PartitionConsumerIterator::new();
-            iterator.set_inner(Box::pin(stream));
-            Ok(iterator)
-        } else {
-            Err(FluvioError::Other(CLIENT_NOT_FOUND_ERROR_MSG.to_owned()))
+        let client = self
+            .inner
+            .as_ref()
+            .ok_or_else(|| FluvioError::Other(CLIENT_NOT_FOUND_ERROR_MSG.to_string()))?;
+
+        let stream = client.stream(offset.0).await?;
+        let mut iterator = PartitionConsumerIterator::new();
+        iterator.set_inner(Box::pin(stream));
+        Ok(iterator)
+    }
+}
+
+pub struct RecordJS {
+    inner: Option<Record>,
+}
+
+#[node_bindgen]
+impl RecordJS {
+    #[node_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self { inner: None }
+    }
+
+    fn set_inner(&mut self, inner: Record) {
+        self.inner = Some(inner);
+    }
+
+    #[node_bindgen]
+    pub fn key(&self) -> Option<String> {
+        self.inner
+            .as_ref()
+            .unwrap()
```

Comment:
I think you might be able to use `?` here.

Reply:
Thanks for the suggestion! That's much cleaner.